### PR TITLE
JetBrains prebuilds doc

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

This PR explains how to enable warming up JB products in prebuilds via experimental support in .gitpod.yml.

## Related Issue(s)

Related to https://github.com/gitpod-io/gitpod/issues/6740

## How to test

- [Indexing for IntelliJ IDEA using Prebuilds](https://deploy-preview-1945--gitpod-kumquat.netlify.app/docs/ides-and-editors/intellij#indexing-using-prebuilds)
- [Indexing for GoLand using Prebuilds](https://deploy-preview-1945--gitpod-kumquat.netlify.app/docs/ides-and-editors/goland#indexing-using-prebuilds)
- [Indexing for PhpStorm using Prebuilds](https://deploy-preview-1945--gitpod-kumquat.netlify.app/docs/ides-and-editors/phpstorm#indexing-using-prebuilds)
- [Indexing for PyCharm using Prebuilds](https://deploy-preview-1945--gitpod-kumquat.netlify.app/docs/ides-and-editors/pycharm#indexing-using-prebuilds)

## Release Notes

```release-note
NONE
```

## Documentation

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/1945"></a>

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/1945"></a>

### CODE REVIEW

File: gitpod/docs/ides-and-editors/goland.md [hunk 1]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
...
+jetbrains:
+  goland:
+    prebuilds:
+      version: stable
+```
+
+- push changes to your Git repository to apply.
+
+`verison` property allows to control whether you would like to index only for `stable`, `latest` or `both` versions.
+You can switch between `stable` and `latest` version of {title} in [your user preferences](https://gitpod.io/preferences).
```
Comment (loujaybee) [inline gitpod/docs/ides-and-editors/goland.md:70]:
```suggestion
The `version` property allows you to control the indexing of JetBrains IDEs which are compatible with Gitpod. You can choose whether to index `stable`, `latest` or `both` IDE versions. Users can switch between `stable` and `latest` versions of {title} on the [user preferences](https://gitpod.io/preferences) page.
```

File: gitpod/docs/ides-and-editors/goland.md [hunk 2]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
 </figure>
 
 **Important:** You must restart any started workspaces for your IDE preferences to take effect.
+
+## Warming up in Prebuilds
```
Comment (loujaybee) [inline gitpod/docs/ides-and-editors/goland.md:45]: Are we sure we want to adopt the "warming" terminology in Gitpod docs? I don't think we currently talk about warming elsewhere in the docs? 🤔  Maybe we should stick to prebuilds only here? WDYT?

```suggestion
## Indexing a JetBrains IDE using Prebuilds
```

File: gitpod/docs/ides-and-editors/goland.md [hunk 3]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
...
+
+To leverage it:
+
+- configure prebuilds for your repository, refer to [Prebuilds](/docs/prebuilds);
+- enable {title} warming up in prebuilds by editing [.gitpod.yml](/docs/config-gitpod-file) in your repository:
+
+```yaml
+jetbrains:
+  goland:
+    prebuilds:
```
Comment (loujaybee) [inline gitpod/docs/ides-and-editors/goland.md:63]: Now I'm wondering whether "prebuilds" is the right term here ...

As this is enabling indexing, which is leveraging prebuilds? 

But this functionality is unrelated to things like the `init` command and core prebuild functionality? 🤔

Comment (akosyakov) [reply gitpod/docs/ides-and-editors/goland.md:63]: We can reconsider it later for sure. Indexing also is important part of warming up, but not a single, later we will also pre install plugins and so on.

For now I was aligning it with how we configure in other places like `github.prebuilds`: https://www.gitpod.io/docs/references/gitpod-yml#github

We can consider later to remove it from .gitpoy.ml and move to project settings instead on the dashboard.

File: gitpod/docs/ides-and-editors/goland.md (hunk 2 above)
Comment (akosyakov) [reply gitpod/docs/ides-and-editors/goland.md:45]: I can rename it to Indexing, but it is only a part of warming up, albeit most important. Space is talking about warming up as well: https://www.jetbrains.com/help/space/set-up-a-dev-evnvironment.html#step-2-warming-up-the-dev-environment

Comment (akosyakov) [reply gitpod/docs/ides-and-editors/goland.md:45]: warm-up:
>  a period of time when the IDE resolves project dependencies, builds indexes, and does other background activities

Comment (akosyakov) [reply gitpod/docs/ides-and-editors/goland.md:45]: I will rephrase introduction that it is not only about indexing.

**Comment by andreafalzetti** [comment]
It's not part of the changes that you introduced, but I've spotted that in all pages we have the same typo `direclty` instead of `directly`

File: gitpod/docs/ides-and-editors/goland.md [hunk 4]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
 </figure>
 
 **Important:** You must restart any started workspaces for your IDE preferences to take effect.
+
+## Warming up in Prebuilds
+
+> **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6740).
+
+When you open the project {title} starts indexing to load modules and
```
Comment (andreafalzetti) [inline gitpod/docs/ides-and-editors/goland.md:49]:
```suggestion
When you open the project, {title} starts indexing to load modules and
```

File: gitpod/docs/ides-and-editors/phpstorm.md [hunk 5]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
 </figure>
 
 **Important:** You must restart any started workspaces for your IDE preferences to take effect.
+
+## Warming up in Prebuilds
+
+> **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6740).
+
+When you open the project {title} starts indexing to load modules and
+enable the core funtionality like code completion and navigation.
```
Comment (andreafalzetti) [inline gitpod/docs/ides-and-editors/phpstorm.md:50]:
```suggestion
enable the core functionality like code completion and navigation.
```

File: gitpod/docs/ides-and-editors/pycharm.md [hunk 6]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
 </figure>
 
 **Important:** You must restart any started workspaces for your IDE preferences to take effect.
+
+## Warming up in Prebuilds
+
+> **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6740).
+
+When you open the project {title} starts indexing to load modules and
+enable the core funtionality like code completion and navigation.
```
Comment (andreafalzetti) [inline gitpod/docs/ides-and-editors/pycharm.md:50]:
```suggestion
enable the core functionality like code completion and navigation.
```

File: gitpod/docs/ides-and-editors/goland.md [hunk 7]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
...
+enable the core funtionality like code completion and navigation.
```
Comment (andreafalzetti) [inline gitpod/docs/ides-and-editors/goland.md:50]:
```suggestion
enable the core functionality like code completion and navigation.
```

**Comment by akosyakov** [comment]
@andreafalzetti we auto generate docs, it is enough to mention one page 🙏

File: gitpod/docs/ides-and-editors/goland.md [hunk 8]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
...
+```yaml
+jetbrains:
+  goland:
+    prebuilds:
+      version: stable
+```
+
+- push changes to your Git repository to apply.
+
+`verison` property allows to control whether you would like to index only for `stable`, `latest` or `both` versions.
```
Comment (andreafalzetti) [inline gitpod/docs/ides-and-editors/goland.md:69]:
```suggestion
`version` property allows to control whether you would like to index only for `stable`, `latest` or `both` versions.
```

File: gitpod/docs/ides-and-editors/pycharm.md [hunk 9]
```diff
@@ -41,3 +41,30 @@ Gitpod enables code editing using JetBrains {title} currently via [JetBrains Gat
...
+```yaml
+jetbrains:
+  pycharm:
+    prebuilds:
+      version: stable
+```
+
+- push changes to your Git repository to apply.
+
+`verison` property allows to control whether you would like to index only for `stable`, `latest` or `both` versions.
```
Comment (andreafalzetti) [inline gitpod/docs/ides-and-editors/pycharm.md:69]:
```suggestion
`version` property allows to control whether you would like to index only for `stable`, `latest` or `both` versions.
```

**Comment by andreafalzetti** [comment]
> @andreafalzetti we auto generate docs, it is enough to mention one page 🙏

alright, sorry for the spam 😂

**Comment by andreafalzetti** [comment]
Overall, I was able to understand the concept and how to leverage the functionality 👍

**Comment by felladrin** [comment]
After fixing the typos mentioned by @andreafalzetti, it's good to go, in my opinion.
Thanks for writing it up!
